packages/code-snippet/src/CodeSnippetManager.ts

```diff
@@ -25,37 +25,25 @@ export interface ICodeSnippet {
 }
 
 export class CodeSnippetManager {
-  readonly codeSnippetEndpoint = 'api/metadata/code-snippets';
-
   async findAll(): Promise<ICodeSnippet[]> {
-    const getCodeSnippets: Promise<ICodeSnippet[]> = new Promise(
-      (resolve, reject) => {
-        const allCodeSnippets: ICodeSnippet[] = [];
-        SubmissionHandler.makeGetRequest(
-          this.codeSnippetEndpoint,
-          'code snippets',
-          (response: any) => {
-            const codeSnippetsResponse = response['code-snippets'];
-
-            for (const codeSnippetKey in codeSnippetsResponse) {
-              const jsonCodeSnippet = codeSnippetsResponse[codeSnippetKey];
-              const codeSnippet: ICodeSnippet = {
-                name: jsonCodeSnippet.name,
-                displayName: jsonCodeSnippet.display_name,
-                description: jsonCodeSnippet.metadata.description,
-                language: jsonCodeSnippet.metadata.language,
-                code: jsonCodeSnippet.metadata.code
-              };
-              allCodeSnippets.push(codeSnippet);
-            }
-            resolve(allCodeSnippets);
-          }
-        );
-      }
+    const codeSnippetsResponse = await FrontendServices.getMetadata(
+      'code-snippets'
     );
-    const codeSnippets = await getCodeSnippets;
+    const allCodeSnippets: ICodeSnippet[] = [];
+
+    for (const codeSnippetKey in codeSnippetsResponse) {
+      const jsonCodeSnippet = codeSnippetsResponse[codeSnippetKey];
+      const codeSnippet: ICodeSnippet = {
+        name: jsonCodeSnippet.name,
+        displayName: jsonCodeSnippet.display_name,
+        description: jsonCodeSnippet.metadata.description,
+        language: jsonCodeSnippet.metadata.language,
+        code: jsonCodeSnippet.metadata.code
```

Comment:
Hmm.  Should we have the server _model_ promote the various metadata-level properties to the same as `name` and `description`?  If I'm reading these correctly, you could then have `jsonCodeSnippet` implement `ICodeSnippet`.

Reply:
This was written by @karlaspuldaro originally with @lresende help. Iiuc I believe `ICodeSnippet` was written to capture what the UI currently needs, but the back end schema (what the api returns) was written with the expectation of adding more metadata in the future and includes fields we may not yet use in the UI.

Reply:
Ok - if there's no advantage to removing the `metadata` level, I'm good.  This is the kind of thing we need to do prior to 1.0.  I suppose there's some advantage to keeping the metadata-specific properties contained.